File: code/10-state-space-search/c/knapsack_demo.c
#include "state_space_search.h"
#include <stdio.h>

int main(void) {
    const ss_item items[] = {{2,40}, {5,30}, {10,50}, {5,10}};
    ss_knapsack_result result;
    if (ss_knapsack_bnb(items, 4, 16, &result) != SS_OK) return 1;
    printf("profit: %lld\n", (long long)result.profit);
    printf("weight: %lld\n", (long long)result.weight);
    return 0;
}

File: code/10-state-space-search/java/KnapsackDemo.java
public final class KnapsackDemo {
    public static void main(String[] args) {
        KnapsackBranchAndBound.Item[] items = {
            new KnapsackBranchAndBound.Item("A", 2, 40),
            new KnapsackBranchAndBound.Item("B", 5, 30),
            new KnapsackBranchAndBound.Item("C", 10, 50),
            new KnapsackBranchAndBound.Item("D", 5, 10),
        };
        KnapsackBranchAndBound.Result result = new KnapsackBranchAndBound().solve(items, 16);
        System.out.println("profit: " + result.profit);
        System.out.println("weight: " + result.weight);
    }
}
